### ISSUE TITLE
proposal to improve and clarify person roles in relation to text types (work, expression, manifestation, transcription)

### ISSUE THREAD

**Issue by jeffreycwitt** [scta/scta-ontology#1]
## Overview

This is a proposal to improve and clarify person roles in relation to text types (work, expression, manifestation, transcription)

* sctap:author
  * Domain
    * sctar:expression
    * sctar:manifestation
    *  sctar:transcription
  * Range
    * sctar:person

At present the property `role:AUT` is applied indiscriminately to all text subclasses. 

This proposal suggest restricting the domain of role:AUT (or equivalent property) to sctar:expression alone, while creating **new roles** for describing how other Persons/Agents may be involved with a texts production at the sctar:work and sctar:manifestation and scta:transcription level 

### Use Cases (Anticipated uses cases that this change would support)

* We would like to be able to request all texts that have a manifestation edited by a particular person
* We would like to be able to request all transcriptions to which a particular person contributed to. 
* We would like to be able distinguish Persons who are editors of Manifestations and Persons who are editors of Transcriptions.
* We would like to select all expressions, who belong to a work created/inspired by a given author.
  * a modern example of this might be, get all expressions of any text by Herman Melville. Such a search should return both the novel Moby Dick authored by Melville but also Screen Plays written by authors (or co-authored by authors) other than Melville.
* We would like to be select translations of a text by particular translators. 

## Basic Concrete Proposal. 

For the first iteration of this proposal, we are thinking of small modifications by introducing a limited set of new roles. 

* sctap:creator
  * Domain
    * sctar:work
  * Range
    * sctar:person

* sctap:author
  * Domain
    * sctar:expression
  * Range
    * sctar:person

* sctap:editor (at the present: this would cover both editors, manuscript scribes, and translators;)
  * Domain
    * sctar:manifestation
  * Range
    * sctar:person

* sctap:teditor (teditor = transcription editor, person(s) with editorial responsibility over the final transcription)
  * Domain
    * sctar:transcription
  * Range
    * sctar:person

* sctap:transcriber (any one involved with the transcription, xml encoding, data preparation of transcription file; this could also be covered by "contributor")
  * Domain
    * sctar:transcription
  * Range
    * sctar:person

* sctap:contributor (a generic role assigned to contributors working at the margins, generally people make small contributions, assisting someone in one of the main roles described above )
  * Domain
    * sctar:work
    * sctar:expression
    * sctar:manifestation
    * sctar:transcription
  * Range
    * sctar:person

## Requirements for implementation

* We need to run a search across all downstream client applications, checking SPARQL queries an the use of role:aut and role:editor
  * We need to make a list of any query using this property and make sure they would be compatible with the new system. 
  * Any query not compatible with the changes would need to be flagged. 
  * Once the new schema is merged to master branch, flagged queries will need to be immediately updated. 
  * NOTE: this is not the ideal workflow. 
  * Ideally, this schema will be versioned. Then we would have multiple rdf-builders and SPARQL endpoints running, supporting different versions. 
  * Versioned client applications would indicate what SCTA SPARQL endpoint they support and would then point the application to the corresponding SPARQL endpoint. 
  * This would enable controlled movement. 
  * We don't imagine supporting SPARQL endpoints for every version indefinitely, but this would allow staged/planned depreciation, allow us to give client applications notices of depreciation, end of life support, and time to both test against updated versions and time to update.

* New properties would need to be recorded in markdown docs and .ttl docs
* Finally, the SCTA-RDF-Builder would need to be updated to support the new schema. 
  * NOTE: Again ideally the Builder is versioned so that version 1.0 of the builder could claim support for rdf-schema 0.1.0 (and continue to import any 0.1.X version, while version 1.1 of the builder could support version 0.2.X of the schema.)

I'm interested in the communities thoughts. 

@CharlotteFeidicker and @uzahnd how well might this simple change expand our ability to model the Bullinger tradition. @uzahnd, I know it will not cover all cases or capture all nuances. @CharlotteFeidicker and I are thinking about small incremental steps.

**Comment by jeffreycwitt** [comment]
Another property could be "hasPublisher" which I believe would only apply to the Manifestation level. 

Though even here I wonder if this would apply better to the ManifestationCodex rather than the ManifestationText

For example

If 16th century printer creates a codex or volume containing a text of Wodeham and a text of Mair, I suppose it is possible that each text could have a different editor, but if they are included in the same physical codex/volume, would it ever be the case that they had different Publishers?? @uzahnd ?? 

I don't think so, which suggests to me that this is a property (hasPublisher) better assigned to the ManifestationCodex while the hasEditor is better assigned to the Manifestion or (ManifestationText)

**Comment by jeffreycwitt** [comment]
Another issue that we need to consider is at what structural levels these properties level. 

Should every paragraph be allowed to take these properties, or just the "item" level or even just the top level. 

In the latter case, clients would be expected to query from a paragraph to its containing Item or topLevelCollection and then infer the author, editor, etc from this resource. 

This would reduce database redundancy but make queries more complicated.

**Comment by CharlotteFeidicker** [comment]
It seems like the Prefix changes for Properties (sctap) and Classes (sctar). We should discuss if rather this is useful or we should keep the same Namespace (including the same Prefix) for all entities of the ontology. If the distinction was made to provide better human - readability concerning the difference between Properties and Classes, we might use upper-case-first-letters for Classes and lower-case-first-letters for Properties.

**Comment by CharlotteFeidicker** [comment]
I have some questions to clarify/discuss : 

- Can every Individual rdf:type sctar:Textitem rather be rdf:type sctar:Manifestation or rdf:type sctar:Work or rdf:type sctar:Expression but not more than one of these? Are sctar:Manifestation, sctar:Work and sctar:Expression therefore owl:disjointWith (each other)?

- Is every Individual rdf:type sctar:Transcription always also one of sctar:Manifestation or sctar:Expression or sctar:Work? Or is every Individual rdf:type sctar:Transcription always rdf:type 
sctar:Manifestation?

- Has every Manifestation an Expression of a Work? Would we need than Properties like "has expression", "has work", "has transcription"?

- Is a hierarchy given, following this schema: Every paragraph <p>[...]</p> is part of a manifestation of a higher ranked expression that expresses a higher ranked work?  

- Are all these question irrelevant for the reason that we don't want to express the relationship between several textitems in this ontology but the relationships between textitems and persons?

**Comment by jeffreycwitt** [comment]
started progress with adding "editor" property. 
edf-schema: ae7ba73d2c1db06f98f67b24a4947398e330aa0b allows placement of editor field within top level witnesses/manifestations
scta-rdf-builder: 3a7e69a83a8106fa96d1b93906ae0feca856a547 adds `editor` property to top level manifestation
lbpweb-js
0bf9e9de0e35d72d50cf11f43c4be8622b2e2582 adds support for grabbing editor off of the toplevel manifestation and displays nicely in citation component.

I think scta-rdf-builder should also place "editor" at structure item level and the lbp-webjs should be adjusted again to grab editor from structureItem. Editor could change from item to item, and this would allow more flexibility.
